support-frontend/assets/pages/contributions-landing/contributionsLandingActions.ts

```diff
@@ -397,6 +405,15 @@ const selectAmount =
 		}))(dispatch, getState);
 	};
 
+const selectAmounts =
+	(amounts: SelectedAmounts) =>
+	(dispatch: Dispatch, getState: () => State): void => {
+		setFormSubmissionDependentValue(() => ({
+			type: 'SELECT_AMOUNTS',
+			amounts,
+		}))(dispatch, getState);
+	};
+
```

Comment:
I find this really hard to read but I think it's a bigger chunk of work to refactor this pattern. I've just copied the `selectAmount` action and tweaked it. I'll have a go at explaining it for clarity though.

`selectAmounts` is a Thunk creator. The body of the created thunk then calls another Thunk creator `setFormSubmissionDependentValue` (which happens to take a function as it's input) and then immediately invokes the newly created Thunk. 

I think we can definitely improve/simplify this, but that's a bigger job!